minio_storage/storage.py

```diff
@@ -236,6 +235,7 @@ def listdir(self, path: str) -> T.Tuple[T.List, T.List]:
         try:
             objects = self.client.list_objects(self.bucket_name, prefix=path)
             for o in objects:
+                assert o.object_name is not None
```

Comment:
Structurally, a Minio `Object.object_name` could be `None`, which it is when the `Object` represents a "directory". It's slightly annoying that we need this assertion, but I like it more than just outright ignoring the type.